report/latest_map_generated.go

```diff
@@ -198,6 +198,21 @@ func (m StringLatestMap) DeepEqual(n StringLatestMap) bool {
 	return true
 }
 
+// EqualUnderMerge should return true if m.Merge(n) is the same as m.
+// We interpret this as "ignoring timestamps", i.e. if all keys and
+// values are the same that's good enough.
+func (m StringLatestMap) EqualUnderMerge(n StringLatestMap) bool {
+	if m.Size() != n.Size() {
+		return false
+	}
+	for i := range m {
+		if m[i].key != n[i].key || m[i].Value != n[i].Value {
+			return false
+		}
+	}
+	return true
+}
```

Comment:
The code is indeed exactly `DeepEqual` ignoring timestamps.
It would be more complicated to cope with the case `m.Size() < n.Size()`, and that never happens in practice.
Do you think a comment like "Implemented conservatively: returns false in some situations where a stricter implementation would return true" would help?

Reply:
> Do you think a comment like "Implemented conservatively: returns false in some situations where a stricter implementation would return true" would help?

Indeed it would! :) Actually I think renaming to `DeepEqualIgnoringTimestamps` is what I'd prefer since that's exactly what the method does, but adding a comment is also fine.

I think also adding the sentence _It would be more complicated to cope with the case m.Size() < n.Size(), and that never happens in practice._ that you wrote above as a comment would be of no harm, since otherwise it might not be obvious why the implementation was conservative.

Reply:
I changed the name to `EqualIgnoringTimestamps`.